Userland/Utilities/acpi.cpp

```diff
@@ -0,0 +1,116 @@
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
+#include <LibCore/System.h>
+#include <stdio.h>
+#include <unistd.h>
+
+// https://uefi.org/specs/ACPI/6.4/05_ACPI_Software_Programming_Model/ACPI_Software_Programming_Model.html#system-description-table-header
+struct [[gnu::packed]] SDTHeader {
+    char sig[4];
+    u32 length;
+    u8 revision;
+    u8 checksum;
+    char oem_id[6];
+    char oem_table_id[8];
+    u32 oem_revision;
+    u32 creator_id;
+    u32 creator_revision;
+};
+
+static void pkg_length(AK::Detail::ByteBuffer<32> const& data, u32& next_in_block, u32& next_block)
+{
+    u32 pos = next_block;
+
+    outln("");
+    out("data: ");
+    if (pos > 3) {
+        for (auto i = -3; i < 0; i++)
+            out("{:#02x} ", data[pos + i]);
+    } else
+        out("               ");
+    out("  ");
+    for (auto i = 0; i < 5; i++)
+        out("{:#02x} ", data[pos + i]);
+    outln("");
+
+    u32 length = 0;
+    u32 delta = 0;
+    int const size = data[pos + 1] >> 6;
+    if (size == 0) {
+        length = data[pos + 1] & 0x3f;
+        delta = 1;
+    } else if (size == 1) {
+        length = data[pos + 1] & 0x0f;
+        length = length | (data[pos + 2] << 4);
+        delta = 2;
+    } else if (size == 2) {
+        length = data[pos + 1] & 0x0f;
+        length = length | (data[pos + 2] << 4);
+        length = length | (data[pos + 3] << 12);
+        delta = 3;
+    } else {
+        length = data[pos + 1] & 0x0f;
+        length = length | (data[pos + 2] << 4);
+        length = length | (data[pos + 3] << 12);
+        length = length | (data[pos + 4] << 20);
+        delta = 4;
+    }
+    length -= (delta - 1);
+
+    outln("pos: {} {} {}", delta, pos + delta + length, data.size());
+    next_in_block = next_block + 1 + delta;
+    next_block = next_block + delta + length;
+}
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio rpath"));
+
+    StringView filename;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(filename, "Name of DSDT table", "DSDT table", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
+
+    auto the_file = TRY(Core::Stream::File::open_file_or_standard_stream(filename, Core::Stream::OpenMode::Read));
+
+    TRY(Core::System::pledge("stdio"));
+
+    Array<u8, sizeof(SDTHeader)> header_buffer;
+    Bytes header_span = header_buffer.span();
+    TRY(the_file->read(header_span));
+    if (header_span.size() != sizeof(SDTHeader)) {
+        outln("failed to read SDTHeader from {}", filename);
+        return -1;
+    }
+
+    SDTHeader const* const header = reinterpret_cast<SDTHeader*>(header_buffer.data());
+
+    auto data = TRY(the_file->read_all());
+
+    if ((header->length - sizeof(SDTHeader)) != data.size()) {
+        outln("bad data size, should be {} but is {}", header->length - sizeof(SDTHeader), data.size());
```

Comment:
Should be a `warnln` at the very least. Ideally an `Error::from_string_view`, but that would probably make the string formatting not work. Also, I think we usually capitalize the first character of messages like this.

Reply:
I went with warnln, will experiment with Error::from_string_view before next PR